pkg/cmd/cli/cmd/edit.go

```diff
@@ -187,23 +199,30 @@ func (o *EditOptions) RunEdit() error {
 		if err != nil {
 			return preservedFile(err, file, o.out)
 		}
-		if bytes.Equal(original, edited) {
+		// Compare content without comments
+		uncommentedOriginal, originalErr := yaml.ToJSON(original)
+		uncommentedEdited, editedErr := yaml.ToJSON(edited)
```

Comment:
@smarterclayton if there is any syntax error, the transformation will fail here. Right now I am ingoring this error so we will always compare commented content in case of syntax errors. Thus the need for a custom comments removal utility. Thoughts?